src/word_debt_bot/game/core.py

```diff
@@ -55,3 +55,25 @@ def add_debt(self, player_id: str, amount: int):
         state = self._state
         state[player_id].word_debt += amount
         self._state = state
+
+    def create_leaderboard(self, sort_by: str, lb_len: int):
+        if sort_by not in ["debt", "cranes"]:
+            raise ValueError("ordering is done by 'debt' or 'cranes'")
+        if lb_len < 1:
+            raise ValueError("requested leaderboard length must be 1 or greater")
+        # Make a sort key and sort users
+        if sort_by == "debt":
+            key = lambda u: u[1].word_debt
+        elif sort_by == "cranes":
+            key = lambda u: u[1].cranes
+        users = sorted(self._state.items(), key=key, reverse=True)
+
+        # Produce leaderboard string to return
+        lb = ""
+        i = 1
+        for u in users:
+            lb += f"{i}. {u[1].display_name} - {u[1].word_debt:,} debt - {u[1].cranes:,} cranes\n"
```

Comment:
Safety: Can we also add a guard against exceeding the 2000 character limit? Either by putting a limit on the number of users that can be requested at all, or by checking the length of the result before appending a new line to make sure we won't go over.

Reply:
added:
```python
if len(lb + entry) > 2000:
                break
```
But I am working on testing it's functionality.

Reply:
Ok, verified it works properly (was able to send 56 users with a request of 80).